src/main/java/uk/gov/hmcts/reform/et/syaapi/service/ManageCaseRoleService.java

```diff
@@ -392,4 +432,253 @@ private List<CaseDetails> getCasesByCaseDetailsListAuthorizationAndCaseUserRole(
         }
         return caseDetailsListByRole;
     }
+
+    /**
+     * Revokes the claimant solicitor role from a case identified by the given submission reference.
+     *
+     * <p>This method performs the following steps:
+     * <ul>
+     *     <li>Logs the intention to revoke the claimant solicitor role for the specified submission reference.</li>
+     *     <li>Retrieves the {@link CaseDetails} where the current user holds the "creator" role.</li>
+     *     <li>If the case is found, revokes the "[CLAIMANT_SOLICITOR]" role assigned to that case.</li>
+     *     <li>Removes the claimant representative data from the case.</li>
+     * </ul>
+     * If the case is not found, a {@link ManageCaseRoleException} is thrown.</p>
+     *
+     * @param authorisation the authorization token of the user performing the operation
+     * @param caseSubmissionReference the unique reference used to locate the case
+     * @return the updated {@link CaseDetails} object with the claimant representative removed
+     * @throws IOException if there is an error during retrieval or revocation of user roles
+     * @throws ManageCaseRoleException if no matching case is found for the provided reference
+     */
+    public CaseDetails revokeClaimantSolicitorRole(String authorisation, String caseSubmissionReference)
+        throws IOException {
+        log.info("Revoke claimant solicitor role for case submission reference: {}", caseSubmissionReference);
+        CaseDetails caseDetails =
+            getUserCaseByCaseUserRole(authorisation, caseSubmissionReference, CASE_USER_ROLE_CREATOR);
+        if (ObjectUtils.isEmpty(caseDetails)) {
+            log.error("Unable to find case details for case submission reference: {}",
```

Comment:
Just thinking if we're throwing the exception and logging it out elsewhere, do we need this log, it will effectively 2x the logs if this occurs